spanner/row.go

```diff
@@ -378,3 +389,186 @@ func (r *Row) ToStructLenient(p interface{}) error {
 		true,
 	)
 }
+
+// SelectAll iterates all rows to the end. After iterating it closes the rows,
+// and propagates any errors that could pop up.
+// It expects that destination should be a slice. For each row it scans data and appends it to the destination slice.
+// SelectAll supports both types of slices: slice of structs by a pointer and slice of structs by value,
+// for example:
+//
+//	type Singer struct {
+//	    ID    string
+//	    Name  string
+//	}
+//
+//	var singersByPtr []*Singer
+//	var singersByValue []Singer
+//
+// Both singersByPtr and singersByValue are valid destinations for SelectAll function.
+//
+// Before starting, SelectAll resets the destination slice,
+// so if it's not empty it will overwrite all existing elements.
+func SelectAll(rows Iterator, v interface{}, options ...DecodeOptions) error {
```

Comment:
```suggestion
func SelectAll(rows Iterator, destination interface{}, options ...DecodeOptions) error {
```

(Or clarify in the documentation above that `v` equals `destination`.)